Funcky/Extensions/EnumerableExtensions/Interleave.cs

```diff
@@ -0,0 +1,45 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.Contracts;
+using System.Linq;
+
+namespace Funcky.Extensions
+{
+    public static partial class EnumerableExtensions
+    {
+        /// <summary>
+        /// Interleaves the elements of multiple sequences by consuming the heads of each subsequence in the same order as the given subsequences. This repeats until all the sequences are completley consumed.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements in the source sequences.</typeparam>
+        /// <param name="sequence">first sequence.</param>
+        /// <param name="otherSequences">other sequences.</param>
+        /// <returns>one sequences with all the elements interleaved.</returns>
+        [Pure]
+        public static IEnumerable<TSource> Interleave<TSource>(this IEnumerable<TSource> sequence, params IEnumerable<TSource>[] otherSequences)
+            => ImmutableList.Create<IEnumerable<TSource>>().Add(sequence).AddRange(otherSequences).Interleave();
```

Comment:
There's an overload that creates a list with one item:
```suggestion
            => ImmutableList.Create(sequence).AddRange(otherSequences).Interleave();
```